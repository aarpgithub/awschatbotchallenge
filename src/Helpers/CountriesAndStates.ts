import states = require("us-state-codes");

export const stateNameToCode = (state: string) => {
  let stateCode = states.sanitizeStateName(state);
  if (stateCode == null) {
    throw Error(`Error missing field state.  Input: ${state}`);
  }
  stateCode = states.getStateCodeByStateName(stateCode);
  if (stateCode == null) {
    throw Error("Error converting state to code.  Input: ${state}");
  }
  return stateCode;
};
