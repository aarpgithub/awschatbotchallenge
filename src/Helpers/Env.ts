export class Env {
  static getString(key: string): string {
    return process.env[key] || "";
  }

  static getInt(key: string): Number {
    return parseInt(process.env[key] || "0");
  }
}
