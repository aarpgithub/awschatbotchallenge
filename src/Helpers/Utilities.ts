import { isNil } from "lodash";
import { createHash } from "crypto";
import traverse = require("traverse");
import { union as lodash_union, flatten as lodash_flatten } from "lodash";

/**
 * Remove any keys with undefined or null values from obj and return obj.  
 * !! Mutates argument !!
 * 
 * example: 
 *  const obj = {a: null, b: 1}
 *  const obj2 = excludeNil(obj)
 *  console.log(obj2)
 * -> "{ b: 1}"
 * @param obj 
 */
export function excludeNil<T>(obj: T): T {
  for (const key in obj) {
    if (isNil(obj[key])) {
      delete obj[key];
    }
  }
  return obj;
}

export const sha256 = (str: string): string => {
  const hash = createHash("sha256");
  hash.update(str);
  return hash.digest("hex");
};

export const sha256checksum = (obj: Object): string => {
  return sha256(JSON.stringify(obj));
};

export const removeKeysRecursive = (obj: any, keys: Array<String>) => {
  traverse(obj).forEach(function(e: any) {
    if (this.key && containsKey(this.key, keys)) {
      this.remove(true);
    }
  });
};

export const containsKey = (key: string, keys: Array<String>) => {
  if (keys.indexOf(key) > -1) {
    return true;
  }
  return false;
};

export const snooze = (ms: number) => {
  return new Promise(resolve => setTimeout(resolve, ms));
};

export const union = lodash_union;
export const flatten = lodash_flatten;
