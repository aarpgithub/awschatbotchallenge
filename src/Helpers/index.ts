export * from "./CountriesAndStates";
export * from "./Env";
export * from "./Utilities";
