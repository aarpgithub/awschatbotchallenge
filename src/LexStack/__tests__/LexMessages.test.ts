import {
  createDelegateAction,
  createElicitSlotAction,
  createPlainTextMessage,
  createElicitIntentAction,
  createConfirmAction,
  createCloseAction
} from "../../Lexstack";

import { removeKeysRecursive } from "../../Helpers";

it("Should be able to create a plaintext message", () => {
  const m = createPlainTextMessage("sample message");
  expect(m).toMatchSnapshot();
});
it("Should be able to create a Delegate Action", () => {
  const m = createDelegateAction({
    slots: {}
  });
});
it("Should be able to create a Close Action", () => {
  const m = createCloseAction({
    fulfillmentState: "Fulfilled",
    message: createPlainTextMessage("test")
  });
  expect(m).toMatchSnapshot();
});
it("Should be able to create a Confirm action", () => {
  const m = createConfirmAction({
    // TODO: fix this
    slots: {},
    message: createPlainTextMessage("sample message")
  });
  expect(m).toMatchSnapshot();
});
it("Should be able to create a plaintext message", () => {
  const m = createPlainTextMessage("sample message");
  expect(m).toMatchSnapshot();
});
