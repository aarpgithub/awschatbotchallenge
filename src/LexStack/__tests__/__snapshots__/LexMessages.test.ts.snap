// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be able to create a Close Action 1`] = `
Object {
  "fulfillmentState": "Fulfilled",
  "message": Object {
    "content": "test",
    "contentType": "PlainText",
  },
  "type": "Close",
}
`;

exports[`Should be able to create a Confirm action 1`] = `
Object {
  "message": Object {
    "content": "sample message",
    "contentType": "PlainText",
  },
  "slots": Object {},
  "type": "ConfirmIntent",
}
`;

exports[`Should be able to create a plaintext message 1`] = `
Object {
  "content": "sample message",
  "contentType": "PlainText",
}
`;

exports[`Should be able to create a plaintext message 2`] = `
Object {
  "content": "sample message",
  "contentType": "PlainText",
}
`;
